classes/webservice/WebserviceSpecificManagementAttachments.php

```diff
@@ -0,0 +1,369 @@
+                ),
+                [
+                    1,
+                    Response::HTTP_INTERNAL_SERVER_ERROR,
+                ]
+            );
+        }
+
+        // Physical file location
+        $file = _PS_DOWNLOAD_DIR_ . $attachment->file;
+        // Check if file exists
+        if (!file_exists($file)) {
+            throw new WebserviceException(
+                sprintf(
+                    'Unable to load the attachment file for attachment %d',
+                    $attachmentId
+                ),
+                [
+                    1,
+                    Response::HTTP_INTERNAL_SERVER_ERROR,
+                ]
+            );
+        }
+
+        // Return file details
+        return [
+            'file' => $file,
+            'mime' => $attachment->mime,
+            'file_name' => $attachment->file_name,
+            'file_size' => $attachment->file_size,
+        ];
+    }
+
+    /**
+     * Handles file upload
+     *
+     * Creates new attachment or replaces existing with a new file.
+     * [PUT] update existing attachment file
+     * [POST] create new attachment
+     */
+    public function executeFileAddAndEdit(): void
+    {
+        // Load attachment without checking the method, because of
+        // PUT which is cleared the $_FILES var in multipart/form context
+        $attachmentId = null;
+        if (isset($this->getWsObject()->urlSegment[2])) {
+            $attachmentId = (int) $this->getWsObject()->urlSegment[2];
+        }
+
+        $attachment = new Attachment($attachmentId);
+
+        $maximumSize = ((int) Configuration::get('PS_ATTACHMENT_MAXIMUM_SIZE')) * 1024 * 1024;
+        $uploadedFile = new UploadedFile(
+            _PS_DOWNLOAD_DIR_,
+            $maximumSize
+        );
+
+        if (isset($_FILES['file'])) {
+            // Standard HTTP upload
+            $fileToUpload = $_FILES['file'];
+        } else {
+            // Get data from binary
+            $fileToUpload = file_get_contents('php://input');
+        }
+
+        try {
+            $file = $uploadedFile->upload($fileToUpload);
+            if (!empty($attachment->id)) {
+                unlink(PS_DOWNLOAD_DIR . $attachment->file);
+            }
+
+            $attachment->file = $file['id'];
+            $attachment->file_name = $file['file_name'];
+            $attachment->mime = $file['mime_type'];
+            $attachment->name[Configuration::get('PS_LANG_DEFAULT')] = $_POST['name'] ?? $file['file_name'];
+
+            if (!empty($attachment->id)) {
+                $attachment->update();
+            } else {
+                $attachment->add();
+            }
+            // Remember affected entity
+            $this->attachment_id = $attachment->id;
+        } catch (MaximumSizeExceeded $e) {
+            $this->getWsObject()->errors[] = $this->trans(
+                'The file is too large. Maximum size allowed is: %1$d kB. The file you are trying to upload is %2$d kB.',
```

Comment:
What do you think of: 'The file you are trying to upload is %2$d KB, which is larger than the maximum size allowed of %1$d KB.'